### PR TITLE
Add method to resolve nested SVG elements

### LINKED ISSUE #195: handle nested svg elements
the noto-emoji regional flag of Slovenia ([SV.svg](https://github.com/googlefonts/noto-emoji/blob/svg_flags2/third_party/region-flags/svg/SI.svg)) contains such an `<svg>` element nested inside the root `<svg>`, which establishes a new viewport and user-coordinate system.
Picosvg needs to be taught how to resolve these.

placing these scattered resources below as reminder to self:
https://www.w3.org/TR/SVG/coords.html
https://www.sarasoueidan.com/blog/nesting-svgs/

### PR DESCRIPTION
**Pull request by anthrotype**
Meant to fix #195 

This PR depends and is based off #202 which has to be merged first.

It adds a method to recursively replace all nested SVGs with equivalent `<g>` groups with a `transform` attribute that combines the translation (given by the inner svg's x and y properties) and the scale + translation given by potential new `viewBox` of the inner svg.

It does _not_ do two things that pertain to nested SVGs:
1) resolve percentage units relative to the nearest ancestor SVG viewport
2) apply a clip rectangle the size of the new inner viewport (blocked on #200).

However, this already does fix the issue with the Noto Emoji's Slovenian flag that contains a nested svg element for its emblem (added as test below).

### CODE REVIEW

File: src/picosvg/svg.py [hunk 1]
```diff
@@ -891,6 +889,99 @@ def normalize_opacity(self, inplace=False):
...
+
+        # first recurse to un-nest any nested nested SVGs
+        self._swap_elements(
+            (el, self._unnest_svg(el, viewbox.w, viewbox.h))
+            for el in self._iter_nested_svgs(svg)
+        )
+
+        g = etree.Element(f"{{{svgns()}}}g")
+        for el in svg:
+            g.append(el)
```
Comment (rsheeter) [inline src/picosvg/svg.py:928]: How list-like is g? Can you g.extend(svg)?

File: src/picosvg/svg.py [hunk 2]
```diff
@@ -891,6 +889,99 @@ def normalize_opacity(self, inplace=False):
...
+            g.attrib["transform"] = transform.tostring()
+
+        # TODO Define a viewport-sized clipPath once transform+clip-path issue is fixed
+        # https://github.com/googlefonts/picosvg/issues/200
+        return (g,)
+
+    def resolve_nested_svgs(self, inplace=False):
+        """Replace nested <svg> elements with equivalent <g> with a transform.
+
+        NOTE: currently this is still missing two features:
```
Comment (rsheeter) [inline src/picosvg/svg.py:951]: Should clarify what it does if it encounters the unsupported things: not-quite-right answer or error?

File: src/picosvg/svg_meta.py [hunk 3]
```diff
@@ -170,3 +172,10 @@ def parse_css_declarations(
         elif declaration.strip():
             raise ValueError(f"Invalid CSS declaration syntax: {declaration}")
     return "; ".join(unparsed) + ";" if unparsed else ""
+
+
+def parse_view_box(s: str) -> Rect:
+    box = tuple(float(v) for v in re.split(r",|\s+", s))
+    if len(box) != 4:
+        raise ValueError("Unable to parse viewBox")
```
Comment (rsheeter) [inline src/picosvg/svg_meta.py:180]: suggest inclusion of the string you couldn't parse it from

**Review by rsheeter — APPROVED** [review]

File: src/picosvg/svg.py (hunk 1 above)
Comment (anthrotype) [reply src/picosvg/svg.py:928]: good point, yes extend does work, i'll change it

File: src/picosvg/svg.py (hunk 2 above)
Comment (anthrotype) [reply src/picosvg/svg.py:951]: no error but not-quite-right-answer (we don't clip and don't resolve percents relative to the correct ancestor)

**Pull request merged**